ci/docker/runtime_functions.sh

```diff
@@ -313,16 +313,22 @@ build_ubuntu_amalgamation_min() {
 build_ubuntu_gpu_cmake_mkldnn() {
     set -ex
     cd /work/build
+    # We need to use generic archtecture. Otherwise, MKLDNN compiled in one
+    # CPU architecture (e.g., C5) can't run on another architecture (e.g., g3).
     cmake \
         -DUSE_CUDA=1               \
         -DUSE_CUDNN=1              \
         -DUSE_MKLML_MKL=1          \
         -DUSE_MKLDNN=1             \
         -DCMAKE_BUILD_TYPE=Release \
+        -DARCH_OPT_FLAGS="-mtune=generic" \
         -G Ninja                   \
         /work/mxnet
 
     ninja -v
+    # libmkldnn.so.0 is a link file. We need an actual binary file named libmkldnn.so.0.
+    cp 3rdparty/mkldnn/src/libmkldnn.so.0 3rdparty/mkldnn/src/libmkldnn.so.0.tmp
+    mv 3rdparty/mkldnn/src/libmkldnn.so.0.tmp 3rdparty/mkldnn/src/libmkldnn.so.0
```

Comment:
This is a no-op. Please elaborate

Reply:
This isn't no-op. I think my comment explain this. libmkldnn.so.0 is a link file. We need an actual binary file named libmkldnn.so.0. Jenkins can't pack a link file.

Reply:
I see, so you're making use of the fact that cp automatically resolves the symlink? I think a cleaner way would be to resolve the symlink explicitely and then override the symlink file with the original file rather than relying on implicit actions.

Reply:
https://stackoverflow.com/questions/7665/how-to-resolve-symbolic-links-in-a-shell-script?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa

Reply:
what you provide is a different thing. it tries to figure out the path to a file. Here is to get a regular file from a link file.

Reply:
I don't think so. My linked thread asks how to resolve a symlink. You're basically doing exactly the same with the copy and move. Copy copies the linked file behind the symlink to a temp file and the move overrides the link with the temp file.

Reply:
What we need here is to turn a link file to a regular file, instead of finding the path to the regular file. It's a different thing. Jenkins can't wrapper a link file and mxnet wants a library file named libmkldnn.so.0, instead of libmkldnn.so.0.13. If what you find can turn libmkldnn.so.0 (a link file) to a regular file with the same name with a single command, can you please provide the command?

Reply:
https://stackoverflow.com/a/13396140/3062895

Reply:
so the code will be something like this.
```
cp --remove-destination 3rdparty/mkldnn/src/`readlink 3rdparty/mkldnn/src/libmkldnn.so.0` 3rdparty/mkldnn/src/libmkldnn.so.0bar.pdf
```

do you think this is a preferred way or less confusing way?

Reply:
To be honest, I actually prefer the existing solution over the one-liner.

Reply:
Alright, no strong feelings from my side